firmware/application/apps/ble_app.cpp

```diff
@@ -0,0 +1,403 @@
+        type[currentPacket] = entry_.packetData.data[currentByte++];
+
+        // This should never happen, but in here just in case.
+        // Break because we can't trust rest of data.
+        // if (length[currentPacket] > entry_.packetData.dataLen)
+        // {
+        //     break;
+        // }
+
+        // Subtract 1 because type is part of the length.
+        for (i = 0; i < length[currentPacket] - 1; i++) {
+            data[currentPacket][i] = entry_.packetData.data[currentByte++];
+        }
+
+        currentPacket++;
+    }
+
+    for (i = 0; i < currentPacket; i++) {
+        uint8_t number_data_lines = ceil((float)(length[i] - 1) / 10.0);
+        uint8_t current_line = 0;
+        std::array<std::string, 5> data_strings{{""}};
+
+        for (j = 0; (j < (number_data_lines * 10)) && (j < length[i] - 1); j++) {
+            if ((j / 10) != current_line) {
+                current_line++;
+            }
+
+            data_strings[current_line] += to_string_hex(data[i][j], 2);
+        }
+
+        // Readd the type back to the total length.
+        field_rect = draw_field(painter, field_rect, s, to_string_hex(length[i]), to_string_hex(type[i]) + pad_string_with_spaces(3) + data_strings[0]);
+
+        if (number_data_lines > 1) {
+            for (k = 1; k < number_data_lines; k++) {
+                if (data_strings[k] != "") {
+                    field_rect = draw_field(painter, field_rect, s, "", pad_string_with_spaces(5) + data_strings[k]);
+                }
+            }
+        }
+    }
+}
+
+void BleRecentEntryDetailView::set_entry(const BleRecentEntry& entry) {
+    entry_ = entry;
+    set_dirty();
+}
+
+static std::uint64_t get_freq_by_channel_number(uint8_t channel_number) {
+    uint64_t freq_hz;
+
+    switch (channel_number) {
+        case 37:
+            freq_hz = 2402000000ull;
+            break;
+        case 38:
+            freq_hz = 2426000000ull;
+            break;
+        case 39:
+            freq_hz = 2480000000ull;
+            break;
+        case 0 ... 10:
+            freq_hz = 2404000000ull + channel_number * 2000000ull;
+            break;
+        case 11 ... 36:
+            freq_hz = 2428000000ull + (channel_number - 11) * 2000000ull;
+            break;
+        default:
+            freq_hz = 0xffffffffffffffff;
+    }
+
+    return freq_hz;
+}
+
+void BLERxView::focus() {
+    field_frequency.focus();
+}
+
+BLERxView::BLERxView(NavigationView& nav)
+    : nav_{nav} {
+    baseband::run_image(portapack::spi_flash::image_tag_btle_rx);
+
+    add_children({ &rssi,
+                       &channel,
+                       &field_rf_amp,
+                       &field_lna,
+                       &field_vga,
+                       &options_region,
+                       &field_frequency,
+                       &check_log,
+#if USE_CONSOLE
+                       &console });
+#else
+                       &recent_entries_view,
+                       &recent_entry_detail_view });
```

Comment:
Reminder for me. Hmm, this is passing in a detail_view? -- I need to think on that.

Reply:
Is that still needed?

Reply:
I need to think about a better way to setup this define. Basically make a switch to use console or non-console for debug purposes. However, now that the app is now moreso in a stable place. This probably isn't needed.